### ISSUE TITLE
[bug]: [config modal]: List of unions not displaying spec name anywhere

### ISSUE THREAD

**Issue by chrisguida** [Start9Labs/embassy-os#1459]
steps to reproduce:
1. Pack a service with this config spec:

```
nodes:
  type: list
  subtype: union
  name: Lightning Nodes
  description: List of Lightning Network node instances to manage
  range: "[1,*)"
  default:
  - lnd
  spec:
    display-as: "{{name}}"
    unique-by: name
    name: Node Implementation
    tag:
      id: type
      name: Type
      description: |
        - LND: Lightning Network Daemon from Lightning Labs
        - C-Lightning: C-Lightning from Blockstream
      variant-names:
        lnd: LND
        c-lightning: C-Lightning
    default: lnd
    variants:
      lnd:
        name:
          type: string
          name: Node Name
          description: Name of this node in the list
          default: "Embassy LND"
          nullable: false
```

2. Visit config modal in UI. Notice that the string "Node Implementation" does not appear anywhere, while "Type" appears twice:

<img width="742" alt="Screen Shot 2022-05-25 at 10 19 27 AM" src="https://user-images.githubusercontent.com/7445670/170298659-d6f285f3-d85d-418b-b3a3-6a822681838b.png">

Expected behavior:

Instead of displaying `Type` twice, the config modal should display `Node Implementation` (that is, the spec name rather than the tag name) in place of the top instance of `Type`, which seems like the logical place for the spec name to go.

**Comment by chrisguida** [comment]
Odd, this code hasn't changed since 02x, I could have sworn it displayed properly on there

**Comment by chrisguida** [comment]
I guess the question is, should I simply delete this name key, as it's not being used? Seems very odd to display the tag name twice, while the spec name isn't displayed anywhere (or isn't supported)

**Comment by MattDHill** [comment]
The spec name and the tag name are referring to the same thing. For instance, if you renamed "Type" to "Node Implementation", the same information would be communicated to the user. The problem here is the repetition, it's useless and maybe even a little confusing. I believe this may actually be a UI bug where we accidentally but the tag name as both a header and line item. It should just be the line item as seen below:

<img width="800" alt="Screen Shot 2022-05-31 at 12 54 33 PM" src="https://user-images.githubusercontent.com/9935159/171264087-1671ea3b-211d-4bbf-9aa6-289cb1a2f5b7.png">

**Comment by MattDHill** [comment]
I'm going to make this change and also do some general reformatting of the config menu for clarity.

**Comment by MattDHill** [comment]
Fixed in #1484
